index.d.ts

```diff
@@ -368,7 +355,9 @@ export interface IconSymbolizer extends BasePointSymbolizer {
    */
   keepUpright?: boolean;
   /**
-   * TODO: CHECK where this property came from and if we need it
+   * Property relevant for mapbox-styles.
+   * If true, text will display without their corresponding icons when the icon
+   * collides with other symbols and the text does not.
    */
   optional?: boolean;
```

Comment:
Isn't this the same as in line 280?

Reply:
Almost, as it is the opposite. :)
362 affects the icon, 282 affects the text.
Its only relevant for mapbox styles and if a Text and Icon Symbolizer are present.

Reply:
OK, thanks for the clarification @KaiVolland.